### DISCUSSION TITLE
Battery charge limiter is not working / hope this helps someone else

### DISCUSSION THREAD

**Discussion by oM0NG0o** [seerge/g-helper#361] [Q&A]
#### Battery charge limiter is not working
Open application log.text from ``%AppData%\GHelper`` . 
If you see something like ``BatteryLimit = 60 : OK`` there (with your selected limit). 
App has done everything it could to set a limit. 
It could be that MyASUS or other Asus services are overwriting this limit after. 
You may want to right click and save this [debloat.bat](https://raw.githubusercontent.com/seerge/g-helper/main/debloat.bat) and then right-click Run it As Admin. 
It will stop not mandatory asus services.

The asus Optimization/Customization.ini located here.

C:\ProgramData\ASUS\ASUS System Control Interface\ASUSOptimization\Customization.ini
The Charge Limit Value is not over written by G-Helper. So adjust it!
Under BatteryHealthCharging Value=100 change to same % as g-helper and save.
 #####################
[Main]
legal=1
[ASUSOLEDCare]
Taskbar=0
[BatteryHealthCharging]
version=3
value=100
[TaskFirst]
value=0
[WiFiSmartConnect]
value=0
 #####################

**Comment by seerge** [comment]
@oM0NG0o thats interesting finding :) I have optimization service, but did not had that file. I assume (if you don't want asus service to overwrite) it should be enough just to delete whole entry.

```
[BatteryHealthCharging]
version=3
value=100
```

I have moved this to Discussions and marked as FAQ (as it's not an issue) and more people can see it.

**Reply by Metalogic999** [reply]
I've been having the issue of my battery fully charging to 100%, (though the G-Helper 60% limit seemed to work sometimes) despite a) Having it set to 60% in G-Helper and b) Having both uninstalled MyAsus and then running debloat.bat. BUT I could still see the entry above in C:\ProgramData\ASUS\ASUS System Control Interface\ASUSOptimization\Customization.ini - so I've now commented out (with an # at the start of each line) the whole [BatteryHealthCharging] section. Will have to see if this makes a difference. I'll try to report back with an update once I've let some time elapse.

**Reply by seerge** [reply]
@Metalogic999 i would just delete whole section (if you are afraid - make a backup). + you need to restart asus optimization service after to pick up changes.

**Reply by Metalogic999** [reply]
Fair enough, I'll do that

**Comment by martink777** [comment]
To be fair, I've been having this issue too - Where MyAsus kept my battery at 60% religiously, I keep turning my laptop on and finding that it's fully charged, or sometimes even with GHelper open it still charges - And even when it does work, it will keep the battery at 59% or 62% instead (not a major issue, just odd).

Interestingly in my version of the Customization.ini, under the Battery Charging section the version is set to '2' not '3' like OP - Not sure what difference that makes.

**Reply by seerge** [reply]
@martink777 g-helper sets charge level only on start / wakeup, it doesn't constantly bash it (as it's not needed). So fact that setting changes (from someone else) while app is running means nothing :)

**Reply by Metalogic999** [reply]
I assume the battery limiter should also still apply when the computer is turned off, but plugged in? Wouldn't be very useful otherwise!

**Reply by seerge** [reply]
@martink777 it sets setting, and bios remembers it (even when laptop is off) untill someone else overwrites.
Apps don't run when laptop is off ...

**Comment by seerge** [comment]
Ok, for everyone involved in a topic @oM0NG0o @martink777 @Metalogic999
You can try this build:
next to setting real charge limit in bios app will also adjust that ini file, so myasus / optimization service will also set same value

[GHelper.zip](https://github.com/seerge/g-helper/files/11456405/GHelper.zip)

**Reply by martink777** [reply]
You're an absolute legend, mate.

**Reply by seerge** [reply]
@martink777 the only thing, i'm not sure if optimization service will honor non-standard values (like 50 or 75 or so). Needs to be tested ..

**Reply by martink777** [reply]
I wondered about that, so I've been keeping it at 60% rather than 50% as that's the lowest value MyAsus supported when I had it installed.
It doesn't seem to automatically discharge to reach the set charge level, so running on battery atm and will plug back in once it gets below 60%.

**Reply by Metalogic999** [reply]
Thanks, will give it a go

**Reply by Metalogic999** [reply]
> I wondered about that, so I've been keeping it at 60% rather than 50% as that's the lowest value MyAsus supported when I had it installed. It doesn't seem to automatically discharge to reach the set charge level, so running on battery atm and will plug back in once it gets below 60%.

@martink777 In the version of MyAsus that came with my 2023 G14, it would actually _only_ let you set a value of 80% or 100%, they removed the 60% option for some reason (perhaps too many people forgot to fully charge when they went out and run out of battery too quickly...), so will be interesting to see if it stays around 60%, after going below that and then getting plugged in again.

@seerge I can see that v 0.60 is writing the value from G-Helper to the Asus  .ini file correctly BTW

**Reply by Metalogic999** [reply]
@seerge Looks like it's working for me, it's consistently only charging to 60% :-)

**Reply by seerge** [reply]
Cool, it's live under v0.61 now . I really hope i stop getting annoying complains that "battery charge is not applied correctly" (while it does, but gets overwritten)

**Reply by assoc** [reply]
@seerge It looks like v0.61 will crash when MyASUS is not installed (ex, system software from Windows Update).
Consider checking that [the file](https://github.com/seerge/g-helper/blob/main/app/OptimizationService.cs#L16) actually exists.

**Reply by seerge** [reply]
@assoc good one, i have uploaded a hotfix over existing 0.61 https://github.com/seerge/g-helper/releases/tag/v0.61

**Reply by bel57** [reply]
Hello, quite an interesting discussion here.
@martink777 did it work to let the laptop run on battery for a while and once plugged the battery limiter stopped at 60%? I'm currently in a situation where everything is set at 60% but my battery is fully charged.
So I'm wondering if I just have to do that or if it will go back to 100% instead.
![battery](https://github.com/seerge/g-helper/assets/10614811/36de1246-af08-4658-bdbd-4c32ab05b656)

**Comment by Metalogic999** [comment]
Sadly I still seem to having some issues with this, current charge is showing as 100%, despite having a 60% charge limit set in G-Helper 0.61

Values in the Asus cusomization.ini look like this:

[BatteryHealthCharging]
version=3
value=60

**Reply by seerge** [reply]
@Metalogic999 > current charge is showing as 100% 
showing where? :)

**Reply by Metalogic999** [reply]
It's showing as 100% now when you hover over the battery icon on the bottom right of the Windows task bar.

**Reply by seerge** [reply]
@Metalogic999 but that’s just your current charge, not a set limit…

**Reply by Metalogic999** [reply]
Yes, of course, but it shouldn't have charged all the way to 100%, as my limit is set to 60% in G-Helper (and I can see it's also set to that in the Asus cusotmization.ini) and I've got G-Helper running constantly, so how was it able to charge to 100%?

**Reply by seerge** [reply]
@Metalogic999 i don't know :) but since it charged somehow to 100% it won't discharge cause of the limit anyway. Also it could be simply issues with bios in 2023 models still, same as it has issues with going eco sometimes

**Reply by Metalogic999** [reply]
@seerge Okay, no worries, I just thought you should know that sometimes somehow the limit seems to get bypassed still. Yes I'm aware that once at 100% it won't discharge to the limit, unless I use the laptop unplugged and let it drop below that, then plug in again.

**Comment by erickplc16** [comment]
Hello,

maybe a dumb question but in other Asus laptops they had the option to use laptop to 60% battery charge and then it gets down to 58% and charge again i think its called max duration or something like that, in G-Helper You have the option to change the value but does it will going to work similar to that option?

thanks for your time to reply.

**Reply by seerge** [reply]
@erickplc16 it's just simple command for BIOS that lets you set max charge %, that's it :) there is no way to control other things unfortunately

**Comment by Beygi93** [comment]
i have a Asus TUF A15 2023 - FA507nv (actually china edition : Tianxuan4 ). the model not showing in bottom right corner of G-Helper! :(

**Comment by VisaHolder** [comment]
Would anyone happen to have the debload.bat? Currently experiencing the same issue with charge going above 85%.

**Reply by seerge** [reply]
@reaprrr you don't need debloat.bat, it's embedded in the app for a long time as ``Start / Stop`` services button under ``Extra``

**Reply by imLLUS1ON** [reply]
@reaprrr how'd you solve it?
